includes/components/class-geolocation.php

```diff
@@ -37,6 +37,9 @@ public function __construct( $args = [] ) {
 		// Set models.
 		$this->models = array_intersect( $this->models, (array) get_option( 'hp_geolocation_models', [ 'listing' ] ) );
 
+		// Update settings.
+		add_action( 'hivepress/v1/activate', [ $this, 'update_settings' ] );
```

Comment:
Maybe we can resolve this without an extra callback, e.g. use the alter_settings callback to add another element to the Default Fields "default" value, if it already contains an array and you add another element, then this "default" should exactly be used if you re-activate the plugin, defaults are set on installation/deactivation already.